<launch>

    <node pkg="tf" type="static_transform_publisher" name="camera_transform" args="1.234 -0.0156 0.91 3.14 0.43 0.0 base_link camera_link 100"/>
    <node pkg="tf" type="static_transform_publisher" name="world_transform" args="0 0 0 0 0 0 world_link base_link 100"/>
    <include file="$(find realsense2_camera)/launch/rs_camera.launch"/>

	<node name="find_object_3d" pkg="find_object_2d" type="find_object_2d" output="screen">
		<param name="gui" value="true" type="bool"/>
		<param name="settings_path" value="~/.ros/find_object_2d.ini" type="str"/>
		<param name="subscribe_depth" value="true" type="bool"/>
		<param name="objects_path" value="" type="str"/>
		<param name="object_prefix" value="object" type="str"/>

		<remap from="rgb/image_rect_color" to="camera/color/image_raw"/>
		<remap from="depth_registered/image_raw" to="camera/aligned_depth_to_color/image_raw"/>
		<remap from="depth_registered/camera_info" to="camera/aligned_depth_to_color/camera_info"/>
	</node>

	<!-- Example of tf synchronisation with the objectsStamped message -->
	<!--node name="tf_example" pkg="find_object_2d" type="tf_example" output="screen">
		<param name="map_frame_id" value="/map" type="string"/>
		<param name="object_prefix" value="object" type="str"/>
	</node-->
	<!-- fake some tf frames for the example /map -> /odom -> /base_link -> /camera_link -->
	<!--node pkg="tf" type="static_transform_publisher" name="base_to_camera_tf"
		args="0.1 0.0 0.3 0.0 0.0 0.0 /base_link /camera_link 100" />
	<node pkg="tf" type="static_transform_publisher" name="odom_to_base_tf"
		args="1.0 0.0 0.1 1.5707 0.0 0.0 /odom /base_link 100" />
	<node pkg="tf" type="static_transform_publisher" name="map_to_odom_tf"
		args="0.0 0.5 0.0 0.7853 0.0 0.0 /map /odom 100" /-->
</launch>

